client/landing/stepper/declarative-flow/internals/steps-repository/goals/select-goals.tsx

```diff
@@ -51,7 +55,8 @@ export const SelectGoals: React.FC< SelectGoalsProps > = ( {
 			</div>
 
 			<div className="select-goals__actions-container">
-				<Button primary onClick={ onSubmit }>
+				<ImportLink onClick={ () => onSubmit( [ ...selectedGoals, SiteGoal.Import ] ) } />
```

Comment:
Good point @phcp. Let me look into this a little bit more. 👍

Reply:
Ideally the `navigation.submit` function would return a promise that would resolve once the submit action has completed. 

![Screenshot 2022-06-20 at 20 13 02](https://user-images.githubusercontent.com/2019970/174658796-1ce9ac3e-74d0-45dd-bc5e-5dd432e52dd2.png)

Never-the-less I noticed a work-around in many other steps where they use local state of `busy` to disable multiple, consecutive submissions.  Similarly I have now introduced `submitting` local state of `SelectGoals`. When `submitting` is `true` I will block further goals selection, or import, or `Continue` clicks. 

![Screenshot 2022-06-20 at 20 18 06](https://user-images.githubusercontent.com/2019970/174659069-148227b4-e0aa-49af-b240-1f183b955fd7.png)

The `Button` component comes pre-equipped with `busy` state animation, so at least an user has some idea that something is happening if he was to be stuck at the screen for a little longer. However if `Continue` or `Import your site content` would fail on their first click for some reason the user has no way to re-submit. They can only press wordpress logo or `Skip to Dashboard`, or the old good page re-load. 😐

Reply:
> However if Continue or Import your site content would fail on their first click for some reason the user has no way to re-submit. 

I'm concerned about this one. Letting user pass is better.

From our Slack conversations, we decided to fix duplication and ignore busy state as it is an edge case. Busy state will be removed.

Thanks for fixing the duplication. `new Set()` approach looks neat, and looking at other places in Calypso, it is also used (was just worried to what level of new Set() will be transpiled to in Calypso), but if other places in Calypso are using it, it should be saqfe for us to use.

Reply:
Got it. My concern was about the duplication itself, `new Set()` seems to be a good approach to solve it.